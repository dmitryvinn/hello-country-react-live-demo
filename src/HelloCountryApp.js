import React from "react";
import CountryFlagImage from "./CountryFlagImage";
import WelcomeCountryText from "./WelcomeCountryText";
import ChangeCountryButton from "./ChangeCountryButton";

class HelloCountryApp extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            country: "american",
            label: "America",
            flag: "https://i.imgur.com/c56iDt7.png"
        };
        this.ChangeCountryButton = this.ChangeCountryButton.bind(this);
    }

    ChangeCountryButton() {
        var countryName = this.state.country;
        var CountryFlagImage = this.state.flag;
        var countryLabel = this.label;
        if (countryName === "america") {
            countryName = "america";
            countryLabel = "America";
            CountryFlagImage = "https://i.imgur.com/c56iDt7.png";
        } else {
            countryName = "canada";
            countryLabel = "Canada";
            CountryFlagImage = "https://i.imgur.com/bh3BkeN.png";
        }
        this.setState({
            country: countryName,
            flag: CountryFlagImage,
            label: countryLabel
        });
    }

    render() {
        return (
            <div>
                <WelcomeCountryText name={this.state.label} />
                <CountryFlagImage img={this.state.flag} />
                <div>
                    <ChangeCountryButton action={this.ChangeCountryButton} />
                </div>
            </div>
        );
    }
}

export default HelloCountryApp;
