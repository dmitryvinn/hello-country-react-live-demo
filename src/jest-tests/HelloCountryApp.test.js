import React from 'react';
import renderer from 'react-test-renderer';
import HelloCountryApp from "../HelloCountryApp";

test('Testing entire app', () => {
    // Initialize Component
    let component = renderer.create
    (
        <HelloCountryApp/>
    );
    // Format result
    let tree = component.toJSON();
    // Assert
    expect(tree).toMatchSnapshot();

    // // Fail scenario
    // component.root.findByProps({className: "changeCountry"}).props.onClick();
    // expect(component.toJSON()).toMatchSnapshot();

});