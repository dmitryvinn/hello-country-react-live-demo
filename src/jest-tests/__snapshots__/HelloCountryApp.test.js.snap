// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Change Button  1`] = `
<div>
  <h1>
    Hello, 
    America
  </h1>
  <div />
  <div>
    <button
      className="changeCountry"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#555555",
          "border": "none",
          "color": "#ffffff",
          "display": "inlineBlock",
          "fontSize": "17px",
          "padding": "15px 32px",
          "textAlign": "center",
          "textDecoration": "none",
        }
      }
    >
      Switch  Country
    </button>
  </div>
</div>
`;

exports[`Testing entire app 1`] = `
<div>
  <h1>
    Hello, 
    America
  </h1>
  <div />
  <div>
    <button
      className="changeCountry"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#555555",
          "border": "none",
          "color": "#ffffff",
          "display": "inlineBlock",
          "fontSize": "17px",
          "padding": "15px 32px",
          "textAlign": "center",
          "textDecoration": "none",
        }
      }
    >
      Switch  Country
    </button>
  </div>
</div>
`;

exports[`Testing entire app 2`] = `
<div>
  <h1>
    Hello, 
    America
  </h1>
  <div />
  <div>
    <button
      className="changeCountry"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#555555",
          "border": "none",
          "color": "#ffffff",
          "display": "inlineBlock",
          "fontSize": "17px",
          "padding": "15px 32px",
          "textAlign": "center",
          "textDecoration": "none",
        }
      }
    >
      Switch  Country
    </button>
  </div>
</div>
`;
